src/compute/fixed_msm.rs

```diff
@@ -79,3 +80,32 @@ impl<T: Curve> Drop for MsmHandle<T> {
         }
     }
 }
+
+/// Extend MsmHandle to work with affine coordinates for short Weierstrass curve elements
+pub trait SwMsmHandle {
+    /// Type of an Affine curve element
+    type AffineElement;
+
+    /// Create a handle from affine generators
+    fn new_with_affine(generators: &[Self::AffineElement]) -> Self;
+
+    /// Compute a MSM with the result given as affine elements
+    fn affine_msm(&self, res: &mut [Self::AffineElement], element_num_bytes: u32, scalars: &[u8]);
+}
+
+impl<C: SwCurveConfig + Clone> SwMsmHandle for MsmHandle<ElementP2<C>> {
+    type AffineElement = Affine<C>;
+
+    fn new_with_affine(generators: &[Self::AffineElement]) -> Self {
+        let generators: Vec<ElementP2<C>> = generators.iter().map(|e| e.into()).collect();
+        MsmHandle::new(&generators)
+    }
+
+    fn affine_msm(&self, res: &mut [Self::AffineElement], element_num_bytes: u32, scalars: &[u8]) {
+        let mut res_p: Vec<ElementP2<C>> = vec![ElementP2::<C>::default(); res.len()];
+        self.msm(&mut res_p, element_num_bytes, scalars);
+        for (resi, resi_p) in res.iter_mut().zip(res_p) {
+            *resi = resi_p.into();
+        }
```

Comment:
This might be so cheap where it doesn't matter, but if you `use rayon::prelude::*;`, you can parallelize this with:
```suggestion
        res.par_iter_mut().zip(res_p).for_each(|(resi, resi_p)| {
            *resi = resi_p.into();
        });
```

Reply:
added